docs/host-powershell/README.md

```diff
@@ -0,0 +1,152 @@
+
+In order to have the powershell load context to control assembly loading for the execution of an application,
+the business logic code needs to be extracted out of the entry point assembly and put into a different assembly, say `'Logic.dll'`.
+The entry point `'Main'` method shall do one thing only -- let the powershell load context load `'Logic.dll'` and start the execution of the business logic.
+Once the execution starts this way, all further assembly loading requests will be handled by the powershell load context.
+
+So the above example needs to be altered as follows in a .NET Core application:
+
+```CSharp
+// MyApp.exe
+using System.Management.Automation;
+using System.Reflection;
+
+namespace Application.Test
+{
+    public class Program 
+    {
+        /// <summary>
+        /// Managed entry point shim, which starts the actual program
+        /// </summary>
+        public static int Main(string[] args)
+        {
+            // Application needs to use PowerShell AssemblyLoadContext if it needs to create powershell runspace
+            // PowerShell engine depends on PS ALC to provide the necessary assembly loading/searching support that is missing from .NET Core 
+            string appBase = System.IO.Path.GetDirectoryName(typeof(Program).GetTypeInfo().Assembly.Location);
+            System.Console.WriteLine("\nappBase: {0}", appBase);
+            
+            // Initialize the PS ALC and let it load 'Logic.dll' and start the execution
+            return (int)PowerShellAssemblyLoadContextInitializer.
+                           InitializeAndCallEntryMethod(
+                               appBase,
+                               new AssemblyName("Logic, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null"),
+                               "Application.Test.Logic",
+                               "Start",
+                               new object[] { args });
+        }
+    }
+}
+
+// Logic.dll
+using System;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+
+namespace Application.Test
+{
+    public sealed class Logic
+    {
+        /// <summary>
+        /// Start the actual logic
+        /// </summary>
+        public static int Start(string[] args)
+        {
+            // My business logic code
+            using (PowerShell ps = PowerShell.Create())
+            {
+                var results = ps.AddScript("Get-Command Write-Output").Invoke();
+                Console.WriteLine(results[0].ToString());
+            }
+            return 0;
+        }
+    }
+}
+```
+
+### .NET Core Sample Application
+
+You can find the sample application project `"MyApp"` under [sample-dotnet1.1](./sample-dotnet1.1). 
+To build the sample project, run the following commands:
+
+```powershell
+dotnet restore .\MyApp\MyApp.csproj
+dotnet publish .\MyApp -c release -r win10-x64
+```
```

Comment:
Unfortunately, as [discussed above](https://github.com/PowerShell/PowerShell/pull/3409#issuecomment-288934991), all powershell NuGet packages that have been published so far only contain powershell assemblies built specifically for Windows. We need to publish NuGet packages that contain both Windows and Unix assemblies, but `dotnet pack` doesn't seem to support that.

#3417 is opened to track this.